CHANGELOG.md

```diff
@@ -3,7 +3,9 @@
 * Drop support for Ruby < 2.7
 * Drop support for SQL Server < 2017
 * Drop support for FreeTDS < 1.0
+* No longer provide a 32-bit Windows build
 * Raise error if FreeTDS is unable to sent command buffer to the server
+* Use freetds v1.4.23, libiconv v1.17 and OpenSSL v3.4.0 for Windows builds
```

Comment:
1. extconts sets these versions for all platforms, right? And it's that their statically linked for Windows. But building for other platforms would also use these versions?
2. Any concern doing this large of a version jump for OpenSSL?

Reply:
> extconts sets these versions for all platforms, right? And it's that their statically linked for Windows. But building for other platforms would also use these versions?

If you would build tiny_tds using our "ports" mechanic, you will also get these versions. Our Linux CI build actually also uses these versions. For installations outside of our CI, the discovered freetds version is used, which could be anything.

> Any concern doing this large of a version jump for OpenSSL?

There is a chance that something breaks. As mentioned on the PR, I wanted to make sure that `tiny_tds` generally works when using encrypted connections by adding this new scenario on CI. But I can imagine that there are some configurations that will no longer work with this new OpenSSL version.

Reply:
> That's what most people will be using across all OS's since we go from CI artifact -> rubygems.org, right?

Not exactly.

When you install `tiny_tds` on Linux or Mac OS, Rubygems will discover that we have some C code in the gem and will try to compile it. In order for `tiny_tds` to compile, it tries to discover an installation of `freetds`. This process does not invoke our `ports` mechanic in case `freetds` is missing, but fails instead.

So all people on non-Windows machines will compile `freetds` on their own first, likely according to our [installation instructions](https://github.com/rails-sqlserver/tiny_tds?tab=readme-ov-file#install).

Mid-term I would like to explore if we could provide precompiled versions of `tiny_tds` for all platforms, similar as [Nokogiri did a while back](https://github.com/sparklemotion/nokogiri/releases/tag/v1.11.0). It would simplify the installation for the end-user.

> Is there any other consideration to add to the changelog? Minimum OS version, minimum glibc version, etc?

Don't think so. I still would like to release the next version as v3 just to signal that there might be breaking changes.